src/data/markdown/docs/30 xk6-browser/01 xk6-browser/01 Browser.md

```diff
@@ -7,18 +7,18 @@ excerpt: "xk6-browser: Browser Class"
-| [browser.isConnected](/javascript-api/xk6-browser/browser/isconnected)                         | Indicates whether the WebSocket connection to the browser application is active or not.                         |
-| 🚧 [browser.newBrowserCDPSession()](/javascript-api/xk6-browser/browser/newbrowsercdpsession) | Allows you to access Browser's CDP session and talk directly to the browser application using the CDP protocol. |
-| [browser.newContext([options])](/javascript-api/xk6-browser/browser/newcontext/)               | Creates and returns a new browser context.                                                                      |
-| [browser.newPage([options])](/javascript-api/xk6-browser/browser/newpage)                      | Creates a new Page in a new BrowserContext and returns the page.                                                |
-| 🚧 [browser.on('disconnected')](/javascript-api/xk6-browser/browser/on)                       | Detects events from the browser application.                                                                    |
-| ❌ [browser.startTracing()](/javascript-api/xk6-browser/browser/starttracing)                  | Starts Chromium Tracing for debugging a Page.                                                                   |
-| ❌ [browser.stopTracing()](/javascript-api/xk6-browser/browser/stoptracing)                    | Stops the tracing created by the browser.startTracing() method.                                                 |
-| [browser.version()](/javascript-api/xk6-browser/browser/version)                               | Returns the browser application's version.                                                                      |
+| State | Method                                                                                         | Description                                                                                                     |
+| - | ---------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| ✅ | [browser.close()](/javascript-api/xk6-browser/browser/close)                                   | Closes the browser and all of its pages (if any were opened).                                                   |
+| ✅ | [browser.contexts()](/javascript-api/xk6-browser/browser/contexts)                             | Allows you to access all open browser contexts.                                                                 |
+| ✅ | [browser.isConnected](/javascript-api/xk6-browser/browser/isconnected)                         | Indicates whether the WebSocket connection to the browser application is active or not.                         |
```

Comment:
Yeah, i was thinking the same thing when i decided to add the state column (will need to add an emoji to signify that it's implemented). The issue is with the display if there is no state column:

<img width="305" alt="Without state column" src="https://user-images.githubusercontent.com/1112428/178001708-f01b8f70-6870-43d4-bf03-8e2677e83fe6.png">

<img width="373" alt="With state column" src="https://user-images.githubusercontent.com/1112428/178001753-fbd069a0-65ea-4f44-8b40-98bbe94a73f8.png">



Happy to remove the state column if you prefer that still.

Reply:
I prefer the look of the first screenshot, besides that line break for `newBrowserCDPSession()`. If we can avoid that by wrapping both the icon and link in a `div`, or using [`white-space: nowrap`](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space), I think it would look OK.

I actually like that it offsets the method names, as it's immediately visible which methods might have problems, since they stick out from the rest of the fully implemented ones. Besides, once all methods are fully implemented, we'd presumably drop the "State" column altogether, so we'd end up with pages that have it, and pages that don't.

Reply:
Resolved in: 19b13f672469cabf9508025f93a1e275046d7920